### DISCUSSION TITLE
Speed,security,OOP

### DISCUSSION THREAD

**Discussion by real-ali** [modular/modular#471] [Q&A]
What is the reason for creating this language, does it have any special features other than speed? And how about security, is it oop?

**Comment by gryznar** [comment]
FAQ

**Comment by mojodojodev** [comment]
[Why Mojo?](https://docs.modular.com/mojo/why-mojo.html) has their reasons, short version is that they had all this great compiler technology for AI through MLIR, but no good programming model for it, and instead of just creating a library and adding more complexity to the ecosystem, they wanted to do a complete rethink across the entire stack to simplify everything. In the process they could bring in all the modern useful programming features like Rust/Swift lifetimes for memory safety, Zig's metaprogramming, Go/Swifts simple concurrency model, low level features from C/C++, and extra things that are important today like ergonomic SIMD. They also wanted to be able to leverage the entire Python ecosystem which has basically "won" AI, it will have OOP features like Python yes as it'll be a full superset, but they're still building out features like Polymorphism.
